doc/api.yml

```diff
@@ -0,0 +1,844 @@
+        - name: page_size
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: follower_name
+          in: query
+          required: false
+          description: Filter follower by first name or lastname.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: List of followers from user connected.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  '/follow/{user_id}':
+    post:
+      tags:
+        - Following
+      summary: Follow user by identifier.
+      operationId: followUserById
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '201':
+          description: The followed user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+  '/unfollow/{user_id}':
+    post:
+      tags:
+        - Following
+      summary: Follow user by identifier.
+      operationId: unfollowUserById
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '201':
+          description: The followed user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  ## Posting resources
+  '/posts':
+    get:
+      tags:
+        - Posting
+      summary: Get all posts.
+      operationId: getPosts
+      parameters:
+        - name: page
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/Page'
+        - name: page_size
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: post_category
```

Comment:
feat> We need to be able to filter posts by many categories at a time
@NyH1409, @fenohasinalala> is it okay to have coma separated categories like 

`/posts?categories=it,programming,security,lang&page_size=20`
 
?

Reply:
@YumeT023 yes it s possible by using an array in query parameters like categories=[it, programing,security] after that in repository we use the "in" operator in our database operation in repository, 
![In operator](https://github.com/blogify-app/blogify-api/assets/103512767/d7078d3f-0f1a-4939-a1f1-80cd3b8a3f84)

Reply:
here is an example of spec
![Specs](https://github.com/blogify-app/blogify-api/assets/103512767/c94881c8-59c1-4113-8314-fcb0c465db0c)

Reply:
i m up for it to implement it in backend 😂

Reply:
so i'll make it like that or ? @NyH1409 @YumeT023 @fenohasinalala